### ISSUE TITLE
[plugin] Requests and fixes to Ride and RideObject APIs

### ISSUE THREAD

**Issue by mgovea** [OpenRCT2/OpenRCT2#11717]
TL;DR: I have some requests to expand the plugin API to make a ride price plugin - similar to the idea referenced in #8610

Two items are key:
 - get/set price of a `Ride`
 - get value of a `Ride`
 
Though I could calculate the value on my own if I could:
 - get age of a `Ride` (or construction date, so I can derive from `GameDate`)

I also noticed an issue in `RideObject`:
 - `excitementMultiplier`
 - `intensityMultiplier`
 - `nauseaMultiplier`

...were all erroneously 0 in my experiments. (I tested a Twist and Looping Coaster.)
I found a workaround by copying `RideRatings` from src/openrct2/ride/RideData.cpp - though duplicating code/configuration makes me sad.

I also saw a similar, though less refined and pre-researched, post at https://forums.openrct2.org/topic/5028-how-do-i-getmodify-ride-information/

**OS:** [Windows 10]
**Version:** [0.2.6]
**Commit/Build:** [8aaf8d0]

**Comment by mgovea** [comment]
The API is more complicated than first look though because of things like ride photos. How should multiple prices be handled? Stalls are part of the Ride interface, and they too can sell more than one thing.

**Comment by mgovea** [comment]
Just saw #11712. That should cover getting (and setting?) the price of a `Ride`. It also includes the `buildDate`.

So I suppose there's everything I need to write my plugin once that merges.

Though it'd be nice to have the Ride value so every plugin wouldn't have to copy the logic of `ride_ratings_calculate_value` from https://github.com/OpenRCT2/OpenRCT2/blob/develop/src/openrct2/ride/RideRatings.cpp

**Comment by MegaMech** [comment]
mgovea, you can compile 11712 so you can make your plugin before it's out. (That's what I did)

**Comment by mgovea** [comment]
Thanks @IntelOrca ! #11712 is merges, so this issue is solved.
(The `RideObject` stuff I mentioned is seemingly because of legacy data stuff that appears to not even be set on the C++ side)
